src/node.cc

```diff
@@ -2487,36 +2491,113 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+class DLib {
+ private:
+  void* handle_;
+  std::string filename_;
+  std::string errmsg_;
+  int flags_;
+#ifndef __POSIX__
+  uv_lib_t lib_;
+#endif
+
+ public:
+  DLib(const char* filename, int flags);
+  bool Open();
+  void Close();
+  const char* GetError();
+
+  operator void*() const {;
+    return this->handle_;
+  }
+
+  static const int DEFAULT_FLAGS;
+};
+
+DLib::DLib(const char* filename, int flags)
+    : handle_(nullptr), filename_(filename), flags_(flags) {
+}
+
+const char* DLib::GetError() {
+  return this->errmsg_.c_str();
+}
+
+#ifdef __POSIX__
+
+const int DLib::DEFAULT_FLAGS = RTLD_LAZY;
+
+bool DLib::Open() {
+  this->handle_ = dlopen(this->filename_.c_str(), this->flags_);
+  if (this->handle_ != nullptr)
+    return true;
+
+  const char* errmsg = dlerror();
+  if (errmsg != nullptr)
+    this->errmsg_ = errmsg;
+  return false;
+}
+
+void DLib::Close() {
+  if (this->handle_)
+    dlclose(this->handle_);
```

Comment:
And set `this->handle_ = nullptr;`?

Side note: `this->` isn't necessary and we don't usually use that style in core.  The fact that the field's name ends in an underscore signifies that it's a data member, not a local.